aks-node-controller/app.go

```diff
@@ -95,6 +102,48 @@ func (a *App) Provision(ctx context.Context, flags ProvisionFlags) error {
 	return err
 }
 
+func (a *App) ProvisionWait(ctx context.Context) (string, error) {
+	if _, err := os.Stat(provisionJSONFilePath); err == nil {
+		data, err := os.ReadFile(provisionJSONFilePath)
+		if err != nil {
+			return "", err
+		}
+		return string(data), nil
+	}
+
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return "", fmt.Errorf("failed to create watcher: %w", err)
+	}
+	defer watcher.Close()
+
+	// Watch the directory containing the provision complete file
+	dir := filepath.Dir(provisionCompleteFilePath)
+	err = os.MkdirAll(dir, 0755) // create the directory if it doesn't exist
+	if err != nil {
+		return "", err
```

Comment:
I usually try to wrap all errors from external packages.

Otherwise, you may encounter a cryptic error not searchable in the codebase.